.github/workflows/run_all_frameworks.yml

```diff
@@ -0,0 +1,128 @@
+name: Run All Frameworks
+
+on: pull_request
+
+
+jobs:
+
+  detect_changes:  # Figure out which frameworks need to be evaluated based on which files are changed.
+    name: Detect Changes
+    runs-on: ubuntu-latest
+    outputs:
+      frameworks: ${{ steps.find-required-tests.outputs.frameworks }}
+      skip_baseline: ${{ steps.find-required-tests.outputs.skip_baseline }}
+      skip_evaluation: ${{ steps.find-required-tests.outputs.skip_evaluation }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: pull base branch
+        run: |
+          git fetch --all
+          git checkout -b $GITHUB_BASE_REF origin/$GITHUB_BASE_REF
+          git checkout $GITHUB_HEAD_REF
+      - id: find-required-tests
+        name: Detect Common Changes  # detect if any changes occurred that should trigger all frameworks.
+        run: |
+          changed_files=$(git diff --name-only HEAD..$GITHUB_BASE_REF)
+          common_files="amlb/** "\
+          "frameworks/shared/** "\
+          ".github/workflows/run_all_frameworks.yml "\
+          ".github/runbenchmark/action.yml "\
+          "runbenchmark.py "\
+          "requirements.txt"
+          echo Common files: $common_files
+          echo Changed files: $changed_files
+
+          is_common=1
+          for f in $changed_files
+          do
+            if echo $common_files | grep -e $f >> /dev/null;
+            then
+                echo "File $f detected as common file"
+                is_common=0
+                break
+            fi
+          done
+
+          # Indicates which jobs should be executed (0) or not (1)
+          skip_evaluation=0
+          if [[ $is_common -eq 0 ]];
+          then
+            FRAMEWORKS='["autogluon", "autosklearn", "gama", "h2oautoml", "mlplanweka", "tpot", "constantpredictor", "randomforest", "tunedrandomforest"]'
```

Comment:
I am not aware if we have any explicit list of frameworks we intend to support throughout changes made to the shared codebase. It would probably be good to have this and load it here instead of using hard coded values.

Reply:
yes, we also need to talk about framework's lifecycle soon, e.g. what should we do with "dead" frameworks like `oboe`, `hypertoptsklearn`? If `autoxgboost` still alive?
Also, now that the frameworks point to the most recent release by default, it can break our workflow on each new release on their side, which can be considered both a good or a bad thing.

Reply:
Good point. I propose that we actually run CI with the latest fixed-version definition. We can (and should) run CI on `stable` upgrades as well, but I don't think it should interfere with our general PR workflow.